sdks/node/src/wandb/settings.ts

```diff
@@ -1,45 +1,78 @@
-import {defaultDomain, defaultHost, getUrls} from '../urls';
+import {defaultHost, getUrls} from '../urls';
 import {Netrc} from '../utils/netrc';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import * as os from 'node:os';
+import * as ini from 'ini';
 
-export function getApiKey(host: string): string {
-  let apiKey = process.env.WANDB_API_KEY;
-  if (!apiKey) {
+function getBaseUrlFromConfig(): string | undefined {
+  // Check environment first
+  const envBaseUrl = process.env.WANDB_BASE_URL;
+  if (envBaseUrl) {
+    return envBaseUrl;
+  }
+
+  // Try to read from config file
+  try {
+    const configPath = path.join(os.homedir(), '.config', 'wandb', 'settings');
+    if (fs.existsSync(configPath)) {
+      const config = ini.parse(fs.readFileSync(configPath, 'utf-8'));
+      if (config.default?.wandb_base_url) {
+        return config.default.wandb_base_url;
+      }
+    }
+  } catch (error) {
+    // Silently fail if we can't read the config file
+    // This could happen in restricted environments like Deno
+  }
+
+  return undefined;
+}
+
+function getApiKeyFromNetrc(host: string): string | undefined {
+  try {
     const netrc = new Netrc();
-    apiKey = netrc.entries.get(host)?.password;
+    return netrc.entries.get(host)?.password;
+  } catch (error) {
+    // Silently fail if we can't read the netrc file
+    // This could happen in restricted environments like Deno
```

Comment:
Why fail silently?

Reply:
Again the failure mode is either the file is corrupt or I don't have read permissions for the file.  Since I can't really tell the user how to fix it, failing silently here will ultimately end up in failing with a no credentials error if the user hasn't set an api key in their environment.